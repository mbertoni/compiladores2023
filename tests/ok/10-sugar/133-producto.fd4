let rec prod1 (acum:Nat) (cont:Nat) (x:Nat) (y:Nat) : Nat = 
    print "Im in: "
    (ifz (x - cont) 
            then (print "acum = " acum)
            else (prod1 (acum+y) (cont+1) x y)
    )


let prod (x:Nat) (y:Nat) :Nat = (prod1 0 0 x y)
let prod43:Nat = print "4 x 3 = " (prod 4 3)