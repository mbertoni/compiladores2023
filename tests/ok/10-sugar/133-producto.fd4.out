acum = 12
Im in: 12
Im in: 12
Im in: 12
Im in: 12
Im in: 12
4 x 3 = 12
